app/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\TaskFormRequest;
+use App\Task;
+use Illuminate\Http\Request;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+class TaskController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $tasks = Task::all();
+
+        return view('tasks.index', compact('tasks'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(TaskFormRequest $request)
+    {
+        $task = Task::create([
+            'name' => $request->name,
+        ]);
+        $task->save();
+
+        return redirect(route('tasks.index'))->with('status', trans('msg.The task has been created!'));
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        //
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        try {
+            $task = Task::findOrFail($id);
+        } catch (ModelNotFoundException $e) {
+            return redirect(route('tasks.index'));
+        }
+
+        return view('tasks.edit', compact('task'));
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(TaskFormRequest $request, $id)
+    {
+        try {
+            $task = Task::findOrFail($id);
+            $task->update([
+                'name' => $request->name,
+            ]);
+            $task->save();
+        } catch (ModelNotFoundException $e) {
+            return redirect(route('tasks.index'));
+        }
+
+        return redirect(route('tasks.edit', $task->id))->with('status', trans('msg.The task has been updated!'));
```

Comment:
```suggestion
        return redirect(route('tasks.edit', $task->id))->with('status', trans('msg.update_successful'));
```
chú ý cachs đặt tên key